### DISCUSSION TITLE
How to change the default key position of copilot in neovim?

### DISCUSSION THREAD

**Discussion by askfiy** [community/community#11611] [Copilot Conversations] [answered]
Very disturbing.

the default C-] , M-] , and M-[ are far off to me, can I remap them in another way?

I use lua for configuration.

Thank you very much.

**Comment by mawkler** [comment, marked as answer]
For instance, if you want to remap them to `Alt-.`/`Alt-,` instead:

Using Vimscript:

```vim
imap <M-.> <Plug>(copilot-next)
imap <M-,> <Plug>(copilot-previous)
```

Using Lua (requires Neovim 0.7+):

```lua
vim.keymap.set('i', '<M-.>', '<Plug>(copilot-next)')
vim.keymap.set('i', '<M-,>', '<Plug>(copilot-previous)')
```

Using Lua (requires Neovim 0.5+):

```lua
nvim_set_keymap('i', '<M-.>', '<Plug>(copilot-next)', { noremap = false })
nvim_set_keymap('i', '<M-,>', '<Plug>(copilot-previous)', { noremap = false })
```

Check out `:help copilot-maps` for more info.

**Reply by askfiy** [reply]
thanks ...

**Reply by Marko-Sanchez** [reply]
Currently on neovim v0.9.1, copilot.vim v1.9.1.

copilot-next and copilot-previous don't seem to work anymore for mapping. I am able to select the suggestion but, moving through the suggestion does not seem to work.

Alternatively, using worked:
```lua
vim.keymap.set("i", "<C-j>", "copilot#Next()", {expr=true, silient=true})
vim.keymap.set("i", "<C-k>", "copilot#Previous()", {expr=true, silient=true})
```
![image](https://github.com/community/community/assets/32421124/4c0e25f6-0dfb-43fa-9d29-8f34b59c1a6f)

_Although my current accept, mappings seem to append some extra binaries at the end of the suggestion '<80>@7'. That random value does not seem to appear when using the default <Tab>_

**Reply by samuelabrl** [reply]
Currently using Neovim v0.10.2 and copilot.vim v1.41.0.
I also encountered the issue where Copilot suggestions had extra binaries (<80>@7) appended at the end of the accepted suggestion.

The issue seems to be related to how vim.keymap.set() evaluates the Copilot functions compared to vim.api.nvim_set_keymap. Switching back to the older vim.api.nvim_set_keymap resolves the issue for now. This is a temporary quick fix, but I hope it helps others facing the same problem.

```lua

vim.api.nvim_set_keymap('i', '<C-l>', 'copilot#Accept()', {expr = true, silent = true, noremap = false})

```

**Reply by fautore** [reply]
Are there any plans to fix `copilot-next` `copilot-previous` or `copilot-accept` not working?
What is the "blessed" way to do this?

edit: apparently setting `replace_keycodes=false` works without using `vim.api.nvim_set_keymap`
```lua
vim.keymap.set('i', '<M-.>', 'copilot#Accept()', { expr=true, replace_keycodes = false })
```
